OSX Build error

found following error on building darktable on OSX after updating to XCode 11.4

```
[  6%] Building CXX object src/external/rawspeed/CMakeFiles/rawspeed.dir/src/librawspeed/common/RawImage.cpp.o
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:408:23: error:
      variable 'threads' must have explicitly specified data sharing attributes
  for (int i = 0; i < threads; i++) {
                      ^~~~~~~
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:404:34: note:
      explicit data sharing attribute requested here
#pragma omp parallel for default(none)                                         \
                                 ^
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:409:47: error:
      variable 'height' must have explicitly specified data sharing attributes
    int y_offset = std::min(i * y_per_thread, height);
                                              ^~~~~~
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:404:34: note:
      explicit data sharing attribute requested here
#pragma omp parallel for default(none)                                         \
                                 ^
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:412:33: error:
      variable 'task' must have explicitly specified data sharing attributes
    RawImageWorker worker(this, task, y_offset, y_end);
                                ^~~~
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:404:34: note:
      explicit data sharing attribute requested here
#pragma omp parallel for default(none)                                         \
                                 ^
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:409:33: error:
      variable 'y_per_thread' must have explicitly specified data sharing
      attributes
    int y_offset = std::min(i * y_per_thread, height);
                                ^~~~~~~~~~~~
/Users/martinstraeten/src/darktable/src/external/rawspeed/src/librawspeed/common/RawImage.cpp:404:34: note:
      explicit data sharing attribute requested here
#pragma omp parallel for default(none)                                         \
                                 ^
4 errors generated.
make[2]: *** [src/external/rawspeed/CMakeFiles/rawspeed.dir/src/librawspeed/common/RawImage.cpp.o] Error 1
make[1]: *** [src/external/rawspeed/CMakeFiles/rawspeed.dir/all] Error 2
make: *** [all] Error 2
martinstraeten@Martins-MacBook-Pro build %
```

--- a/cmake/src-dependencies.cmake
+++ b/cmake/src-dependencies.cmake
@@ -67,7 +67,8 @@ if(WITH_OPENMP)
 
   if((CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
       CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0) OR
-     (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"
+     (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND
+      CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.0.3
       # XCode 10 is broken. Maybe XCode 11 will be ok?
      ))
     # See https://bugs.llvm.org/show_bug.cgi?id=35873

